spring-web/src/main/java/org/springframework/web/server/ResponseStatusException.java

```diff
@@ -78,6 +78,7 @@ public ResponseStatusException(int rawStatusCode, @Nullable String reason, @Null
 	public ResponseStatusException(HttpStatusCode status, @Nullable String reason, @Nullable Throwable cause) {
 		super(status, cause);
 		this.reason = reason;
+		setDetail(reason);
 	}
```

Comment:
I would make this delegate to the protected constructor in the same class:
```java
public ResponseStatusException(HttpStatusCode status, @Nullable String reason, @Nullable Throwable cause) {
	this(status, reason, cause, null, null);
}
```
That becomes more clear and is already covered by tests.

Reply:
Done